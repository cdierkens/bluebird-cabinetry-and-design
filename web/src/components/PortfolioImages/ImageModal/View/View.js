import PropTypes from "prop-types";
import React from "react";

const View = ({ views, index, modalProps: { isFullscreen } }) => {
  const { source, alt } = views[index];

  return (
    <div
      className="leading-none relative text-center box-border flex items-center justify-center"
      style={{ height: isFullscreen ? "100vh" : "calc(100vh - 90px)" }}
    >
      <img
        className="h-auto max-h-full max-w-full m-auto select-none"
        src={source.regular}
        alt={alt}
      />
    </div>
  );
};

View.propTypes = {
  views: PropTypes.array,
  index: PropTypes.number,
  modalProps: PropTypes.object.isRequired,
};

export default View;
