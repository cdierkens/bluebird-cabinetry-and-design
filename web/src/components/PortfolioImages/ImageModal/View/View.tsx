import React from "react";
import { CommonProps } from "react-images";
import { CarouselImage } from "../../lib";

const View: React.FC<CommonProps> = ({
  views = [],
  currentIndex = 0,
  modalProps,
}) => {
  const { source, alt } = views[currentIndex] as any as CarouselImage;

  const height = modalProps?.isFullscreen ? "100vh" : "calc(100vh - 90px)";

  return (
    <div
      className="leading-none relative text-center box-border flex items-center justify-center"
      style={{ height }}
    >
      <img
        className="h-auto max-h-full max-w-full m-auto select-none"
        src={source.regular}
        alt={alt ?? undefined}
      />
    </div>
  );
};

export default View;
