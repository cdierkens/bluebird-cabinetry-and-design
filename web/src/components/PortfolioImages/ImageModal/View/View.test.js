import { render } from "@testing-library/react";
import React from "react";
import View from "./View";

it("matches snapshot", () => {
  const { asFragment } = render(
    <View
      isFullScreen={false}
      modalProps={{}}
      index={0}
      views={[{ source: "", alt: "" }]}
      label="mock label"
    />
  );

  expect(asFragment()).toMatchSnapshot();
});
