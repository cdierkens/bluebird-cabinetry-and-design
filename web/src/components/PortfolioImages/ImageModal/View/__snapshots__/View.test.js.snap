// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="leading-none relative text-center box-border flex items-center justify-center"
    style="height: calc(100vh - 90px);"
  >
    <img
      alt=""
      class="h-auto max-h-full max-w-full m-auto select-none"
    />
  </div>
</DocumentFragment>
`;
